presto-pinot/src/main/java/io/prestosql/pinot/PinotConfig.java

```diff
@@ -0,0 +1,435 @@
+    {
+        this.estimatedSizeInBytesForNonNumericColumn = estimatedSizeInBytesForNonNumericColumn;
+        return this;
+    }
+
+    @NotNull
+    public String getServiceHeaderParam()
+    {
+        return serviceHeaderParam;
+    }
+
+    @Config("pinot.service-header-param")
+    public PinotConfig setServiceHeaderParam(String serviceHeaderParam)
+    {
+        this.serviceHeaderParam = serviceHeaderParam;
+        return this;
+    }
+
+    @NotNull
+    public String getCallerHeaderValue()
+    {
+        return callerHeaderValue;
+    }
+
+    @Config("pinot.caller-header-value")
+    public PinotConfig setCallerHeaderValue(String callerHeaderValue)
+    {
+        this.callerHeaderValue = callerHeaderValue;
+        return this;
+    }
+
+    @NotNull
+    public String getCallerHeaderParam()
+    {
+        return callerHeaderParam;
+    }
+
+    @Config("pinot.caller-header-param")
+    public PinotConfig setCallerHeaderParam(String callerHeaderParam)
+    {
+        this.callerHeaderParam = callerHeaderParam;
+        return this;
+    }
+
+    public boolean isPreferBrokerQueries()
+    {
+        return preferBrokerQueries;
+    }
+
+    @Config("pinot.prefer-broker-queries")
+    public PinotConfig setPreferBrokerQueries(boolean preferBrokerQueries)
+    {
+        this.preferBrokerQueries = preferBrokerQueries;
+        return this;
+    }
+
+    public boolean isForbidSegmentQueries()
+    {
+        return forbidSegmentQueries;
+    }
+
+    @Config("pinot.forbid-segment-queries")
+    public PinotConfig setForbidSegmentQueries(boolean forbidSegmentQueries)
+    {
+        this.forbidSegmentQueries = forbidSegmentQueries;
+        return this;
+    }
+
+    @Nullable
+    public String getRestProxyServiceForQuery()
+    {
+        return restProxyServiceForQuery;
+    }
+
+    @Config("pinot.rest-proxy-service-for-query")
+    public PinotConfig setRestProxyServiceForQuery(String restProxyServiceForQuery)
+    {
+        this.restProxyServiceForQuery = restProxyServiceForQuery;
+        return this;
+    }
+
+    public boolean isUseDateTrunc()
+    {
+        return useDateTrunc;
+    }
+
+    @Config("pinot.use-date-trunc")
+    public PinotConfig setUseDateTrunc(boolean useDateTrunc)
+    {
+        this.useDateTrunc = useDateTrunc;
+        return this;
+    }
+
+    public int getNumSegmentsPerSplit()
+    {
+        return this.numSegmentsPerSplit;
+    }
+
+    @Config("pinot.num-segments-per-split")
+    public PinotConfig setNumSegmentsPerSplit(int numSegmentsPerSplit)
+    {
+        checkArgument(numSegmentsPerSplit > 0, "Number of segments per split must be more than zero");
+        this.numSegmentsPerSplit = numSegmentsPerSplit;
+        return this;
+    }
+
+    public boolean isIgnoreEmptyResponses()
+    {
+        return ignoreEmptyResponses;
+    }
+
+    @Config("pinot.ignore-empty-responses")
```

Comment:
Why would pinot return empty responses?

Reply:
Removed: the default is false and there are no use cases in the code or tests where it's set to true.